23_multi-assay_analyses.Rmd

```diff
@@ -158,6 +160,120 @@ ggplot(correlation_table, aes(x = Var1, y = Var2, fill = cor)) +
   labs(x = "Taxa", y = "Metabolites")
 ```
 
+## Multi-Omics Factor Analysis
+
+Multi-Omics Factor Analysis [@Argelaguet2018] (MOFA) is
+an unsupervised method for integrating multi-omic data sets in a downstream analysis.
+It could be seen as a generalization of principal component analysis. Yet, with the ability to infer a latent (low-dimensional) representation, 
+shared among the mutliple (-omics) data sets in hand.
+
+We use the R [MOFA2](https://biofam.github.io/MOFA2/index.html) package
+for the analysis, and [install](https://biofam.github.io/MOFA2/installation.html) the corresponding dependencies.
+
+**Note:** Comment out the line for installing `miniconda` if it has been installed earlier.
+Similarly for `mofapy2` if it was so.
+
+```{r, message=FALSE, warning=FALSE}
+if(!require(MOFA2)){
+    BiocManager::install("MOFA2")
+}
+
+# For inter-operability between Python and R, and setting Python dependencies,
+# reticulate package is needed
+if(!require(reticulate)){
+    install.packages("reticulate")
+}
+
+reticulate::install_miniconda(force = TRUE)
+reticulate::use_miniconda()
+reticulate::py_install(packages = c("mofapy2"), pip = TRUE)
+```
+
+The `mae` object could be used straight to create the MOFA model:
+
+```{r, message=FALSE, warning=FALSE}
+library(MOFA2)
+# For simplicity, classify all high-fat diets as high-fat, and all the low-fat 
+# diets as low-fat diets
+colData(mae)$Diet <- ifelse(colData(mae)$Diet == "High-fat" | 
+                              colData(mae)$Diet == "High-fat + XOS", 
+                            "High-fat", "Low-fat")
+
+# Removing duplicates at the microbiome data
+# which are also in form e.g. "Ambiguous" and "uncultured" taxa
+mae[[1]] <- mae[[1]][!duplicated(rownames(assay(mae[[1]]))), ]
```

Comment:
Are these rows also otherwise identical? If yes, can you add an issue to mia, proposing a new function to remove duplicate rows? Like `removeDuplicateRows(tse)`

Reply:
I ran into this problem before, and the reason comes from the `mia::agglomerateByRank` when it is used without `onRankOnly=TRUE` then it produces duplicates. e.g.
![image](https://user-images.githubusercontent.com/86964562/149948604-ff955d8b-6a3e-47b9-82a0-0227a0a103a3.png)

Earlier in the chapter the agglomeration was done to Genus without the `onRankOnly` argument.
Shall I change the agglomeration done earlier in the chapter, or still open an issue?

Reply:
yes.

Reply:
Should that  "Ambiguous" and "uncultured" taxa be kept?

Reply:
hmmhmm.. for example purposes, you can remove.